src/resources/InsightPanelConfigs/InsightPanelConfigInterfaces.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * Defines the options to list Insight Panel configurations.
+ */
+export interface InsightPanelConfigListOptions {
+    /**
+     * The zero-based page to retrieve.
+     *
+     * @default `0`
+     */
+    page?: number;
+    /**
+     * The number of configurations to return per page.
+     *
+     * @default `10`
+     */
+    perPage?: number;
+    /**
+     * A substring that must appear in a search interface configuration name for this configuration to appear in results.
+     */
+    filter?: string;
+    /**
+     * The sort order to apply on the Insight Panel configuration name.
+     *
+     * **Allowed values:**
+     * - `asc`: Sort configurations by name in ascending order.
+     * - `desc`: Sort configurations by name in descending order.
+     *
+     * @default `asc`
+     */
+    order?: 'asc' | 'desc';
+}
+
+/**
+ * Defines the parameters required to create an Insight Panel configuration.
+ */
+export interface InsightPanelConfigCreationParams {
+    /**
+     * The ID of the Insight Panel configuration.
+     */
+    id: string;
+
+    /**
+     * The name of the Insight Panel configuration.
+     */
+    name: string;
+
+    /**
+     * The ID of the associated query pipeline.
+     */
+    pipeline: string;
+
+    /**
+     * The context field mappings to use when performing queries. This mapping allows an Insight Panel integration to retrieve the context information from the CRM and pass it to the Coveo platform when executing Insight Panel queries.
+     *
+     * The key is the context information such as `subject` or `description`.
+     *
+     * The value is a CRM reference allowing the Insight Panel integration to retrieve the context information from the CRM.
+     *
+     * @example
+     * ```json
+     * {
+     *   "subject": "Case_subject",
+     *   "description": "Case_description"
+     * }
+     * ```
+     */
+    contextFields?: Record<string, string>;
```

Comment:
These aren't optional in the service.

Reply:
Also, they aren't in the service yet, but they will be _very_ soon.

Reply:
`contextFields` is maked as optional simply to allow a caller to create a config by passing this object.

```json
{
  "name": "my config",
  "pipeline": "my pipeline ID"
}
```

In this case, the service would initialize the context fields with the default value.

Reply:
On the other hand, passing an empty object to `contextFields` should create the config with an empty map because it was explicitly requested by the caller.

Reply:
Sounds good. I misunderstood when I added it to the document. I adjusted it.